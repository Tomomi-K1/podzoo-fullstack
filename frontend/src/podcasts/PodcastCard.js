import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import handleImageError from "../common/handleImageError";
// Material UI
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';


function PodcastCard({podcast}){
    
    // const handleImageError = (e) => {
    //     e.target.onerror = null;
    //     e.target.src = fallbackImg;
    // }
     
    return(
        // how do I keep image ratio 1:1?
        
        <Card sx={{maxWidth: "200px", m:2}} raised>
            <Link to={`/podcast/${podcast.feedId}`}>
            {/* <CardContent> */}
            <CardMedia
                component="img"
                width= "100%"
                height="auto"
                image={podcast.artwork}
                alt="podcast artwork"
                onError={handleImageError}
            />
            {/* </CardContent> */}
                <CardContent>
                    <Typography variant="body1" color="text.secondary" sx={{fontWeight:'bold'}}>
                        {podcast.title}
                    </Typography>
                    <Typography variant="body2" color="text.secondary">
                        {podcast.author}
                    </Typography>
                </CardContent>
            </Link>
        </Card>
        
    )
}

export default PodcastCard;