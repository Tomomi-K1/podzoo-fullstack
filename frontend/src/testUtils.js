import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };