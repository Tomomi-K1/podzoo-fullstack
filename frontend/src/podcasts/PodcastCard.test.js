import React from "react";
import { render, screen, waitFor, debug} from "@testing-library/react";
import { MemoryRouter } from "react-router";
import * as rrd from 'react-router-dom';
import PodcastCard from "./PodcastCard";
import { UserProvider } from "../testUtils";

const podcast ={
    author:'test-author',
    title:'test-title',
    artwork:'test-artwork',
    feedId: 1
}

it('renders', () => {
    render(
        <MemoryRouter>
                <PodcastCard podcast={podcast}/>
        </MemoryRouter>
    ) 
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <PodcastCard podcast={podcast}/>
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('renders podcast information', () => {
    render(
        <MemoryRouter>
                <PodcastCard podcast={podcast}/>
        </MemoryRouter>
    ) 
    expect(screen.getByText('test-title')).toBeInTheDocument();
})


