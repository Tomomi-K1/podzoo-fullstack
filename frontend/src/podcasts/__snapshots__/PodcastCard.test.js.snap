// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i59dwk-MuiPaper-root-MuiCard-root"
  >
    <a
      href="/podcast/1"
    >
      <img
        alt="podcast artwork"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="auto"
        src="test-artwork"
        width="100%"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1vweose-MuiTypography-root"
        >
          test-title
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-cq90jg-MuiTypography-root"
        >
          test-author
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
